server/src/test/java/keywhiz/service/resources/admin/SecretsResourceIntegrationTest.java

```diff
@@ -43,16 +47,49 @@
     keywhizClient = TestClients.keywhizClient();
   }
 
+  @Test
+  public void createsSecretWithOwner() throws Exception {
+    login();
+
+    String owner = UUID.randomUUID().toString();
+
+    keywhizClient.createGroup(owner, "description", ImmutableMap.of());
+
+    SecretDetailResponse response = keywhizClient.createSecret(
+        "name",
+        owner,
+        "description",
+        "content".getBytes(StandardCharsets.UTF_8),
+        ImmutableMap.of(),
+        0);
+    assertEquals(owner, response.owner);
+  }
+
+  @Test
+  public void createSecretWithUnknownOwnerFails() throws Exception {
+    login();
+
+    String owner = UUID.randomUUID().toString();
+
+    assertThrows(IOException.class, () -> keywhizClient.createSecret(
+        "name",
+        owner,
+        "description",
+        "content".getBytes(StandardCharsets.UTF_8),
+        ImmutableMap.of(),
+        0));
+  }
+
   @Test public void listsSecrets() throws IOException {
-    keywhizClient.login(DbSeedCommand.defaultUser, DbSeedCommand.defaultPassword.toCharArray());
+    login();
     assertThat(keywhizClient.allSecrets().stream().map(SanitizedSecret::name).toArray())
```

Comment:
The rest of the changes in this file aren't relevant, I just wanted to replace the login statement with something much more brief to improve readability.